\documentclass[a4paper, 11pt]{book}

\newcommand{\tw}[1]{\textcolor{red}{\bf [TW:~#1]}}
\newcommand{\twcomment}[1]{\textcolor{red}{\bf [TW (COMMENT):~#1]}}

\usepackage{algorithm,algpseudocode}
\usepackage{emptypage}

%\usepackage{fontspec}
\usepackage{lmodern}
\usepackage{lipsum}
\usepackage{psl-cover}
\graphicspath{{Pictures/}} % Specifies the directory where pictures are stored
\usepackage[parfill]{parskip}
\usepackage{graphicx}				% Use pdf, png, jpg, or eps¬ß with pdflatex; use eps in DVI mode
%\usepackage{subfig}
								% TeX will automatically convert eps --> pdf in pdflatex
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{natbib}

%\usepackage{filecontents}
\usepackage{here}
\usepackage{multirow}
\usepackage{tikz}
\usetikzlibrary{arrows.meta}
\usetikzlibrary{shapes.geometric}
\usetikzlibrary{arrows, arrows.meta, shapes.geometric, chains, patterns, shadows, fit, backgrounds}
\usepackage{pgfplots, pgfplotstable}

\tikzstyle{block} = [rectangle, draw, fill=blue!20, node distance=3cm,
    text width=5em, text centered, rounded corners, minimum height=4em]
\tikzstyle{line} = [draw, -latex']

\usepackage{listings}% http://ctan.org/pkg/listings
\lstset{
  basicstyle=\ttfamily,
  mathescape}

\usepackage{hyperref}
\hypersetup{urlcolor=blue, colorlinks=true}

\usepackage{rotating}

%\usepackage{subcaption}
\usepackage{subfig}
\usepackage{xcolor}
\newcommand\myworries[1]{\textcolor{red}{#1}}
\usepackage[TS1,T1]{fontenc}
%\usepackage{fourier, heuristica}
\usepackage{fancyvrb}
\usepackage{array, booktabs}
%\usepackage[x11names]{xcolor}
\usepackage{colortbl}
\usepackage{caption}
\usepackage{bbm}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{mathtools}
\usepackage{float}
\usepackage[autostyle]{csquotes}
\usepackage{multirow}
\usepackage{soul}
\usepackage{longtable}

\usepackage{array}
\newcolumntype{C}[1]{>{\centering\arraybackslash}p{#1}}

\DeclareCaptionFont{blue}{\color{LightSteelBlue3}}
\DeclareMathOperator*{\argmax}{\arg\!\max}

\usepackage{listings}
\definecolor{mygreen}{rgb}{0,0.6,0}
\definecolor{mygray}{rgb}{0.5,0.5,0.5}
\definecolor{mymauve}{rgb}{0.58,0,0.82}

    \makeatletter
    \renewcommand\part{%
       \if@openright
         \cleardoublepage
       \else
         \clearpage
       \fi
       \thispagestyle{empty}%
       \if@twocolumn
         \onecolumn
         \@tempswatrue
       \else
         \@tempswafalse
       \fi
       \null\vfil
       \secdef\@part\@spart}
    \makeatother

\lstset{ %
  backgroundcolor=\color{white},   % choose the background color
  basicstyle=\footnotesize,        % size of fonts used for the code
  breaklines=true,                 % automatic line breaking only at whitespace
  captionpos=b,                    % sets the caption-position to bottom
  commentstyle=\color{mygreen},    % comment style
  escapeinside={\%*}{*)},          % if you want to add LaTeX within your code
  keywordstyle=\color{blue},       % keyword style
  stringstyle=\color{mymauve},     % string literal style
}

\lstset{
  basicstyle=\ttfamily,
  columns=fullflexible,
  showstringspaces=false,
  commentstyle=\color{gray}\upshape
numbers=right, 
                numberstyle=\tiny, 
                breaklines=true,
                numbersep=5pt,
                xleftmargin=.25in,
                xrightmargin=.25in
}

\lstdefinelanguage{XML}
{
  morestring=[b]",
  morestring=[s]{>}{<},
  morecomment=[s]{<?}{?>},
  stringstyle=\color{black},
  identifierstyle=\color{blue},
  keywordstyle=\color{cyan},
  morekeywords={xmlns,version,type}% list your attributes here
}

% \newcommand{\foo}{\color{LightSteelBlue3}\makebox[0pt]{\textbullet}\hskip-0.5pt\vrule width 1pt
% \hspace{\labelsep}}

\newcommand{\p}{\text{p}}
\newcommand{\setsep}{\;\; | \;\;}

\title{Deep learning for computational phenotyping in cell-based assays \\ {\normalfont \fontsize{18}{18}\selectfont Apprentissage profond pour le ph\'enotypage computationnel dans les essais cellulaires}}

\author{Joseph BOYD}

\institute{MINES ParisTech}
\doctoralschool{Ing\'enierie des Syst\`emes, Mat\'eriaux, M\'ecanique, \'Energ\'etique}{621}
\specialty{Bio-informatique}
\date{30 Juin 2020}

%% cotutelle
% \entitle{Thesis Subject in English}
% \otherinstitute{CEA Saclay}
% \logootherinstitute{logo-institute}

\jurymember{1}{Charles KEVRANN}{Directeur de recherche, INRIA Rennes}{Rapporteur}
\jurymember{2}{Anna KRESHUK}{Group leader, EMBL Heidelberg}{Rapporteur}
\jurymember{3}{Auguste GENOVESIO}{Directeur de recherche, INSERM ENS}{Examinateur}
\jurymember{4}{Florian JUG}{Group leader, MPI-CBG}{Examinateur}
\jurymember{5}{Perrine PAUL-GILLOTEAUX}{Ing\'enieur de recherche, CNRS}{Examinateur}
\jurymember{6}{David ROUSSEAU}{Professeur, Universit\'e d'Angers}{Examinateur}
%\jurymember{7}{Veronique STOVEN}{Professeur, MINES Paristech}{Examinateur}
%\jurymember{8}{Thomas WALTER}{Charg\'e de recherche, MINES Paristech}{Directeur de th\`ese}
\jurymember{7}{Thomas WALTER}{Charg\'e de recherche, MINES Paristech}{Examinateur}
\jurymember{8}{Veronique STOVEN}{Professeur, MINES Paristech}{Directeur de th\`ese}
% \jurymember{9}{Pr√©nom NOM}{Titre, √©tablissement}{Invit√©}
% \jurymember{10}{Pr√©nom NOM}{Titre, √©tablissement}{Invit√©}

\frabstract{
Le ph\'enotypage computationnel est un ensemble de technologies Èmergentes permettant d'\'etudier systÈmatiquement le r\^ole du g\'enome dans l'obtention de ph\'enotypes, les caract\'eristiques observables d'un organisme et de ses sous-syst\`emes. En particulier, les essais cellulaires permettent de cribler des panels de petites mol\'ecules ou de moduler l'expression des g\`enes, et de quantifier les effets sur les caract\'eristiques ph\'enotypiques allant de la viabilit\'e \`a la morphologie cellulaire. Le criblage \`a haut contenu \'etend les m\'ethodologies des criblages cellulaires \`a une lecture \`a haut contenu bas\'ee sur des images, en particulier les canaux multiplex\'es de la microscopie \`a fluorescence. Les cribles bas\'es sur de multiples lignÈes cellulaires sont aptes \`a diff\'erencier les ph\'enotypes de diff\'erents sous-types d'une maladie, repr\'esentant l'h\'et\'erog\'en\'eit\'e mol\'eculaire concern\'ee dans la conception de th\'erapies m\'edicales de pr\'ecision. Ces mod\`eles biologiques plus riches sous-tendent une approche plus cibl\'ee pour le traitement de maladies mortelles telles que le cancer. Un d\'efi permanent pour le criblage \`a haut contenu est donc la synth\`ese des lectures h\'et\'erog\`enes dans les cribles ‡ multiples lign\'ees cellulaires. Parall\`element, l'\'etat de l'art \'etabli en mati\`ere d'applications d'analyse d'images et de vision par ordinateur est l'apprentissage profond. Cependant, son r\^ole dans le criblage \`a haut contenu ne fait que commencer \`a \^etre r\'ealis\'e. Cette th\`ese aborde deux probl\'ematiques de l'analyse \`a haut contenu des lign\'ees cellulaires canc\'ereuses. Les contributions sont les suivantes : (i) une d\'emonstration du potentiel d'apprentissage profond et de mod\`eles g\'en\'erateurs dans le criblage \`a haut contenu ; (ii) une solution bas\'ee sur l'apprentissage profond au probl\`eme de l'h\'et\'erog\'en\'eit\'e dans un criblage de m\'edicaments sur plusieurs lign\'ees cellulaires ; et (iii) de nouvelles applications de mod\`eles de traduction d'image \`a image comme alternative \`a la microscopie \`a fluorescence co\^uteuse actuellement n\'ecessaire pour le criblage \`a haut contenu.
}

\enabstract{
Computational phenotyping is an emergent set of technologies for systematically studying the role of the genome in eliciting phenotypes, the observable characteristics of an organism and its subsystems. In particular, cell-based assays screen panels of small compound drugs or otherwise modulations of gene expression, and quantify the effects on phenotypic characteristics ranging from viability to cell morphology. High content screening extends the methodologies of cell-based screens to a high content readout based on images, in particular the multiplexed channels of fluorescence microscopy. Screens based on multiple cell lines are apt to differentiating phenotypes across different subtypes of a disease, representing the molecular heterogeneity concerned in the design of precision medicine therapies. These richer biological models underpin a more targeted approach for treating deadly diseases such as cancer. An ongoing challenge for high content screening is therefore the synthesis of the heterogeneous readouts in multi-cell-line screens. Concurrently, deep learning is the established state-of-the-art image analysis and computer vision applications. However, its role in high content screening is only beginning to be realised. This dissertation spans two problem settings in the high content analysis of cancer cell lines. The contributions are the following: (i) a demonstration of the potential for deep learning and generative models in high content screening; (ii) a deep learning-based solution to the problem of heterogeneity in a multi-cell-line drug screen; and (iii) novel applications of image-to-image translation models as an alternative to the expensive fluorescence microscopy currently required for high content screening.
}

\frkeywords{Apprentissage profond, criblage \`a haut contenu, lign\'ees cellulaires, ph\'enotypage computationnel.}
\enkeywords{Deep learning, high content screening, cell lines, computational phenotyping.}


\begin{document}

\maketitle{}

%\lhead{\emph{Contents}} % Set the left side page header to "Contents"
\tableofcontents % Write out the Table of Contents
%
%\lhead{\emph{List of Figures}} % Set the left side page header to "List of Figures"
\listoffigures % Write out the List of Figures
%
%\lhead{\emph{List of Tables}} % Set the left side page header to "List of Tables"
\listoftables % Write out the List of Tables

\input{Chapters/chapter1}
\input{Chapters/chapter2}
\part{Computational phenotyping for multiple cell lines}
\label{partI}
\input{Chapters/chapter3}
\input{Chapters/chapter4}
\part{Fluorescence-free phenotyping}
\label{partII}
\input{Chapters/chapter5}
\input{Chapters/chapter6}
\input{Chapters/chapter7}

%\addtocontents{toc}{\vspace{2em}} % Add a gap in the Contents, for aesthetics

\appendix % Cue to tell LaTeX that the following 'chapters' are Appendices

% Include the appendices of the thesis as separate files from the Appendices folder
% Uncomment the lines as you write the Appendices

\input{Appendices/appendixA}
\input{Appendices/appendixB}
\input{Appendices/appendixC}
\input{Appendices/appendixD}
%\input{Appendices/appendixE}

\label{Bibliography}

%\lhead{\emph{Bibliography}} % Change the page header to say "Bibliography"

\bibliographystyle{unsrtnat} % Use the "unsrtnat" BibTeX style for formatting the Bibliography

\bibliography{Bibliography} % The references (bibliography) information are stored in the file named "Bibliography.bib"



\end{document}
